Remove code for macOS 13 Ventura and earlier

annevk opened the pull request:
#### a1270a27787057b73b325c32d9aa520b31c758fa
<pre>
Remove code for macOS 13 Ventura and earlier
<a href="https://bugs.webkit.org/show_bug.cgi?id=294524">https://bugs.webkit.org/show_bug.cgi?id=294524</a>

Reviewed by Sam Weinig and Per Arne Vollan.

WebKit supports two versions back (14 and 15).

Canonical link: <a href="https://commits.webkit.org/296269@main">https://commits.webkit.org/296269@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0edd51f1e010d586d15188afcec768236539fb87

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108014 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27679 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18098 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113224 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58531 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109977 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28374 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36228 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81996 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110962 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/22506 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97315 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62428 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21901 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15448 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57971 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/100617 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91842 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15514 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116351 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/106576 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35085 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25829 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91029 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35461 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93593 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90823 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35725 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13480 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/30893 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17444 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34984 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/40538 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/130865 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34727 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35533 "Found 1 new JSC stress test failure: wasm.yaml/wasm/stress/jitless-wasm-to-js-should-follow-to-wasm-call-frame-convention-2.js.default (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38085 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36388 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

weinig commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1073,11 +1067,11 @@
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 130000)
+#if PLATFORM(MAC)
 #define HAVE_UNIFIED_SPEECHSYNTHESIS_FIX_FOR_81465164 1
```
I would guess you could get rid of this altogether and remove the one use.

annevk replied:
The file it's in uses `PLATFORM(COCOA)` which would make the conditional `!PLATFORM(MAC)` instead with a comment that reads `// macOS won't send a did start speaking callback for empty strings.` which seems very confusing. I'm going to leave this as-is for now.

weinig commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1124,14 +1117,9 @@
+#if PLATFORM(COCOA)
 #define HAVE_MEMORY_ATTRIBUTION_VM_SHARE_SUPPORT 1
 #define HAVE_PROHIBIT_PRIVACY_PROXY 1
-#endif
```
These are only used in Cocoa files so can just be removed along with their uses.

weinig commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1124,14 +1117,9 @@
-
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 130000) \
-    || PLATFORM(IOS_FAMILY)
 #define HAVE_MACH_MEMORY_ENTRY_OWNERSHIP_IDENTITY_TOKEN_SUPPORT 1
```
This is only used in Cocoa files so can just be removed along with its uses.

weinig commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1073,11 +1067,11 @@
 #endif
 
-#if HAVE(WEBGPU_IMPLEMENTATION) && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 130000) || PLATFORM(IOS) || PLATFORM(VISION))
+#if HAVE(WEBGPU_IMPLEMENTATION) && (PLATFORM(MAC) || PLATFORM(IOS) || PLATFORM(VISION))
```
For these ones that have most of the platforms now defined, it might be nice to consider instead listing the ones that are explicitly not supported (Catalyst? Simulator? Watch? TV?). Not entirely sure thought and certainly a riskier change.

annevk replied:
Decided to leave as-is for now. I don't know if we want things to magically apply to new platforms or not.

weinig commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1202,17 +1185,15 @@
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 130000) \
-    || PLATFORM(IOS_FAMILY)
+#if PLATFORM(COCOA)
 #define HAVE_DYLD_DLOPEN_IMAGE_HEADER_SPI 1
```
This is only used in darwin files (I think all our support Darwin platforms are also cocoa still) so can just be removed along with its uses.

weinig commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1236,13 +1217,8 @@
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 130000) \
-    || PLATFORM(IOS_FAMILY)
+#if PLATFORM(COCOA)
 #define HAVE_CORE_GRAPHICS_GRADIENT_CREATE_WITH_OPTIONS 1
```
This is only used in CoreGraphics (which is only cocoa now that windows CoreGraphics is not supported) files so can just be removed along with its uses.

weinig commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1236,13 +1217,8 @@
-
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 130000) \
-    || PLATFORM(IOS_FAMILY)
 #define HAVE_CORE_GRAPHICS_PREMULTIPLIED_INTERPOLATION_GRADIENT 1
```
This is only used in CoreGraphics (which is only cocoa now that windows CoreGraphics is not supported) files so can just be removed along with its uses.

weinig commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1263,9 +1238,7 @@
-    && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 130000) \
-    || PLATFORM(IOS_FAMILY))
+#if !defined(HAVE_CGSTYLE_CREATE_SHADOW2) && PLATFORM(COCOA)
 #define HAVE_CGSTYLE_CREATE_SHADOW2 1
```
This is only used in CoreGraphics (which is only cocoa now that windows CoreGraphics is not supported) files so can just be removed along with its uses.

weinig commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1300,17 +1273,14 @@
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 130000)
+#if PLATFORM(MAC)
 #define HAVE_STAGE_MANAGER_NS_WINDOW_COLLECTION_BEHAVIORS 1
```
This is only used in Mac files so can just be removed along with its uses.

weinig commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1300,17 +1273,14 @@
-#endif
-
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 130000)
 #define HAVE_SHARING_SERVICE_PICKER_POPOVER_SPI 1
```
This is unused and can just be removed.

weinig commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1469,7 +1439,7 @@
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 130000)
+#if PLATFORM(MAC)
 #define HAVE_LS_SERVER_CONNECTION_STATUS_RELEASE_NOTIFICATIONS_MASK 1
```
This is only used inside PLATFORM(MAC) so can just be removed along with its uses.

(The use in GPUProcessCocoa.mm is weird and probably broken but is inside a PLATFORM(MAC) block).

weinig commented on Source/WTF/wtf/PlatformUse.h:
```diff
@@ -339,10 +339,6 @@
 #endif
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED < 130000)
-#define USE_AVIF 1
```
You should remove the uses of this as well.

annevk replied:
Other ports still use this through CMake.

weinig replied:
In that case, I think I was trying to have it still listed here with:

```
#if !defined(USE_AVIF)
#define USE_AVIF 0
#endif
```

The goal is have these Platform* files be the definitive place where the macros used throughout the project are all listed.

weinig commented on Source/WebKit/UIProcess/Inspector/mac/WebInspectorUIProxyMac.mm:
```diff
@@ -382,9 +382,7 @@ - (IBAction)_popUpButtonAction:(id)sender
     [window setCollectionBehavior:([window collectionBehavior] | NSWindowCollectionBehaviorFullScreenAllowsTiling)];
 
-#if HAVE(STAGE_MANAGER_NS_WINDOW_COLLECTION_BEHAVIORS)
     [window setCollectionBehavior:([window collectionBehavior] | NSWindowCollectionBehaviorAuxiliary)];
```
This and the line above it can be combined as:

```
    [window setCollectionBehavior:([window collectionBehavior] | NSWindowCollectionBehaviorFullScreenAllowsTiling | NSWindowCollectionBehaviorAuxiliary)];
```

weinig commented on Source/WebKitLegacy/mac/WebCoreSupport/WebInspectorClient.mm:
```diff
@@ -559,9 +559,7 @@ - (NSWindow *)window
     [window setCollectionBehavior:([window collectionBehavior] | NSWindowCollectionBehaviorFullScreenAllowsTiling)];
 
-#if HAVE(STAGE_MANAGER_NS_WINDOW_COLLECTION_BEHAVIORS)
     [window setCollectionBehavior:([window collectionBehavior] | NSWindowCollectionBehaviorAuxiliary)];
```
This and the line above it can be combined as:

```
[window setCollectionBehavior:([window collectionBehavior] | NSWindowCollectionBehaviorFullScreenAllowsTiling | NSWindowCollectionBehaviorAuxiliary)];
```

annevk committed:
Remove code for macOS 13 Ventura and earlier
https://bugs.webkit.org/show_bug.cgi?id=294524

Reviewed by Sam Weinig and Per Arne Vollan.

WebKit supports two versions back (14 and 15).

Canonical link: https://commits.webkit.org/296269@main
